starport/services/network/networkchain/prepare.go

```diff
@@ -205,30 +206,64 @@ func (c Chain) applyGenesisValidators(ctx context.Context, genesisVals []network
 // updateConfigFromGenesisValidators adds the peer addresses into the config.toml of the chain
 func (c Chain) updateConfigFromGenesisValidators(genesisVals []networktypes.GenesisValidator) error {
 	var p2pAddresses []string
-	for _, val := range genesisVals {
-		p2pAddresses = append(p2pAddresses, val.Peer.GetTcpAddress())
+	var tunnelAddresses []TunneledPeer
+	for i, val := range genesisVals {
+		if val.Peer.GetTcpAddress() != "" {
+			p2pAddresses = append(p2pAddresses, val.Peer.GetTcpAddress())
+		} else {
+			tunnel := val.Peer.GetHttpTunnel()
+			addressParts := strings.Split(tunnel.Address, "@")
+			if len(addressParts) != 2 {
+				return errors.Errorf("invalid http tunnel address: %s", tunnel.Address)
+			}
+			tunneledPeer := TunneledPeer{
+				Name:      tunnel.Name,
+				Address:   addressParts[1],
+				NodeID:    addressParts[0],
+				LocalPort: strconv.Itoa(i + 22000),
```

Comment:
Do we really need to have this field in the yaml file, what's the downside of not having?

Reply:
Initially I wanted the code hooked into the start command reads list of tunnelled peers and comes up with local ports by itself. But this way the start command will rewrite config.toml p2p.persistent_peers every run in order to add newly generated local client addresses. So i decided to calculate local ports once during `starport n chain prepare` and write them into p2p.persistent_peers with local ports and add local ports to tunneled peers file. This way start command will use list of tunneled peers only to launch chisel clients and addresses of the local clients will remain the same from run to run. But maybe i chosen not the best way of creating these ports during the prepare command.

Reply:
This makes sense. And this way, users also have chance to overwrite the port here if they have other things running in this port number.

One suggestion, can we rename it to `LocalAddress` and instead of just a port number add the host part as well; `localhost:port_number`?